spec.html

```diff
@@ -129,33 +129,22 @@ <h1>
     <emu-clause id="sec-getiteratorflattenable" type="abstract operation">
       <h1>
         GetIteratorFlattenable (
-          _obj_: an ECMAScript language value,
-          _hint_: ~sync~ or ~async~,
+          _obj_: an ECMAScript language value
         ): either a normal completion containing an Iterator Record or a throw completion
       </h1>
       <dl class="header">
       </dl>
       <emu-alg>
         1. If _obj_ is not an Object, throw a *TypeError* exception.
-        1. Let _alreadyAsync_ be *false*.
-        1. Let _method_ be *undefined*.
-        1. If _hint_ is ~async~, then
-          1. Set _method_ to ? Get(_obj_, @@asyncIterator).
-          1. Set _alreadyAsync_ to *true*.
-        1. If IsCallable(_method_) is *false*, then
-          1. Set _method_ to ? Get(_obj_, @@iterator).
-          1. Set _alreadyAsync_ to *false*.
-        1. If IsCallable(_method_) is *false*, then
+        1. Let _method_ be ? GetMethod(_obj_, @@iterator).
+        1. If _method_ is *undefined*, then
           1. Let _iterator_ be _obj_.
-          1. Set _alreadyAsync_ to *true*.
         1. Else,
           1. Let _iterator_ be ? Call(_method_, _obj_).
         1. If _iterator_ is not an Object, throw a *TypeError* exception.
         1. Let _nextMethod_ be ? Get(_iterator_, *"next"*).
         1. If IsCallable(_nextMethod_) is *false*, throw a *TypeError* exception.
```

Comment:
This line diverges from `GetIterator` but the early failure is IMO better. See related https://github.com/tc39/ecma262/issues/3033.